doc/perception_pipeline/src/cylinder_segment.cpp

```diff
@@ -0,0 +1,354 @@
+  segmentor.setNormalDistanceWeight(0.1);
+  // run at max 1000 iterations before giving up
+  segmentor.setMaxIterations(10000);
+  // tolerance for variation from model
+  segmentor.setDistanceThreshold(0.05);
+  // min max values of radius in meters to consider
+  segmentor.setRadiusLimits(0, 1);
+  segmentor.setInputCloud(cloud);
+  segmentor.setInputNormals(cloud_normals);
+
+  // Obtain the cylinder inliers and coefficients
+  segmentor.segment(*inliers_cylinder, *coefficients_cylinder);
+
+  // Extract the cylinder inliers from the input cloud
+  pcl::ExtractIndices<pcl::PointXYZRGB> extract;
+  extract.setInputCloud(cloud);
+  extract.setIndices(inliers_cylinder);
+  extract.setNegative(false);
+  extract.filter(*cloud);
+}
+
+void cloudCB(const sensor_msgs::PointCloud2ConstPtr& input)
+{
+  // BEGIN_SUB_TUTORIAL callback
+  //
+  // Perception Related
+  // ^^^^^^^^^^^^^^^^^^
+  // First, convert from sensor_msgs to pcl::PointXYZRGB which is needed for most of the processing.
+  pcl::PointCloud<pcl::PointXYZRGB>::Ptr cloud(new pcl::PointCloud<pcl::PointXYZRGB>);
+  pcl::fromROSMsg(*input, *cloud);
+  // Using passthough filter to get region of interest. A passthrough filter just eliminates the point cloud values
+  // which do not lie in the user specified range.
+  passThroughFilter(cloud);
+  // Declare normals and call function to compute point normals.
+  pcl::PointCloud<pcl::Normal>::Ptr cloud_normals(new pcl::PointCloud<pcl::Normal>);
+  computeNormals(cloud, cloud_normals);
+  // inliers_plane will hold the indices of the point cloud that correspond to a plane.
+  pcl::PointIndices::Ptr inliers_plane(new pcl::PointIndices);
+  // Detect and eliminate the plane on which the cylinder is kept to ease the process of finding the cylinder.
+  removePlaneSurface(cloud, inliers_plane);
+  // We had calculated the point normals in a previous call to computeNormals,
+  // now we will be extracting the normals that correspond to the plane on which cylinder lies.
+  // It will be used to extract the cylinder.
+  extractNormals(cloud_normals, inliers_plane);
+  // ModelCoefficients will hold the parameters using which we can define a cylinder of infinite length.
+  // It has a public attribute |code_start| values\ |code_end| of type |code_start| std::vector< float >\ |code_end|\ .
+  // |br|
+  // |code_start| Values[0-2]\ |code_end| hold a point on the center line of the cylinder. |br|
+  // |code_start| Values[3-5]\ |code_end| hold direction vector of the z-axis. |br|
+  // |code_start| Values[6]\ |code_end| is the radius of the cylinder.
+  pcl::ModelCoefficients::Ptr coefficients_cylinder(new pcl::ModelCoefficients);
+  /* Extract the cylinder using SACSegmentation. */
+  extractCylinder(cloud, coefficients_cylinder, cloud_normals);
+  // END_SUB_TUTORIAL
+  if (cloud->points.empty())
+  {
+    ROS_ERROR_STREAM_NAMED("cylinder_segment", "Can't find the cylindrical component.");
+    return;
+  }
+  static bool points_not_found = true;
```

Comment:
rather than use static, make these functions part of a class and use a member variable